ui/client/components/HealthCheck.tsx

```diff
@@ -1,56 +1,161 @@
-      </div>
-    ): null
-  }
+function Content({data}: {data: Omit<HealthCheckResponse, "state">}) {
+  const {t} = useTranslation()
+  return (
+    <span>
+      {data.error || t("healthCheck.unknownState", "State unknown")}
+      {": "}
+      {data.processes?.map((name, index) => (
+        <React.Fragment key={name + index}>
+          {index !== 0 && ", "}
+          <ProcessItem name={name}/>
+        </React.Fragment>
+      ))}
+    </span>
+  )
 }
 
-const mapState = state => ({
-  healthCheckInterval: getFeatureSettings(state)?.intervalSettings?.healthCheck,
-})
+function Collipsable({iconSize, children}: PropsWithChildren<{iconSize: number}>) {
+  const [expanded, setExpanded] = useState<boolean>(false)
+  useEffect(
+    () => {
+      if (expanded) {
+        const timeout = setTimeout(() => setExpanded(e => !e), 5000)
+        return () => clearTimeout(timeout)
+      }
+    },
+    [expanded],
+  )
+  const {ref} = useSize<HTMLDivElement>()
+  const isOverflow = expanded || ref.current?.offsetWidth < ref.current?.scrollWidth
+  const {t} = useTranslation()
+  return (
+    <Transition in={expanded} timeout={500}>
+      {state => (
+        <>
+          <div
+            className={css({
+              display: "flex",
+              flexWrap: "wrap",
+              alignItems: "center",
+              flex: 1,
+              overflow: "hidden",
+              minHeight: iconSize * 2,
+              maxHeight: state === "entering" || state === "entered" ? "60vh" : iconSize * 2,
+              whiteSpace: state === "exited" ? "nowrap" : "normal",
+              transition: "all .25s ease-in-out",
+            })}
+          >
+            <div
+              ref={ref}
+              className={css({
+                overflow: "hidden",
+                textOverflow: "ellipsis",
+                paddingTop: iconSize / 2,
+                paddingBottom: iconSize / 2,
+                paddingRight: iconSize,
+              })}
+            >
+              {children}
+            </div>
+          </div>
+          {(isOverflow || state !== "exited") && (
+            <button
+              className={css({
+                flexShrink: 0,
+                background: "none",
+                border: 0,
+                fontWeight: "bold",
+                textTransform: "uppercase",
+                height: iconSize * 2,
+              })}
+              onClick={() => setExpanded(expanded => !expanded)}
+            >
+              {expanded ? t("healthCheck.hide", "hide") : t("healthCheck.show", "show all")}
+            </button>
+          )}
+        </>
+      )}
+    </Transition>
+  )
+}
 
-type Props = ReturnType<typeof mapState> & EspActionsProps & WithTranslation
+function HealthCheck(): JSX.Element {
+  const [{state, ...data}, setState] = useState<HealthCheckResponse>({state: STATE_OK})
```

Comment:
STATE_OK is not type? I think we should create for state special type like ENUM?